### PR TITLE
Add Bureau position ranking using drag and drop

### PR DESCRIPTION
**Pull request by mjoyce91**
Bureau position bidders can be ranked.

Use with https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/425

- Can drag from unranked to ranked and vice versa
- Can drag to re-rank (only persisted to database for Shortlist, not Candidates)
- Can use dropdown to re-order
- Persisted to database on every change
- Can toggle visibility of shortlist and candidates
- Sort dropdown only affects candidates, not shortlist (shortlist is always ordered by ranking number)

TODO
- [x] Validate in IE11

### CODE REVIEW

File: src/actions/bureauPositionBids.js [hunk 1]
```diff
@@ -42,6 +83,61 @@ export function bureauBidsFetchData(id, query = {}) {
...
+        }),
+      })
+      .then(() => {
+        api()
+          .post('/available_position/ranking/', ranking, {
+            cancelToken: new CancelToken((c) => {
+              cancelRanking = c;
+            }),
+          })
+          .then(({ data }) => data || [])
```
Comment (elizabeth-jimenez) [inline src/actions/bureauPositionBids.js:104]: i think after post we will also want to call bureauBidsRankingFetchData?

**Comment by elizabeth-jimenez** [comment]
Filtering on HS removes the Filter button, so we can't really go back lol.

**Comment by elizabeth-jimenez** [comment]
It seems the list doesn't seem to be 'saving' after we move bidders up. Moving bidders to short list and then clicking 'Go Back' doesn't seem to save the list. I think it's that we are saving the list, but we aren't 'using' the updated one(Post comment above).

Scenario: Make an update to the Short List and then Sort. Bidders that you just moved will go back to unRanked. However, if you 'Go Back' and come back into the Position Details, it works as we want.

File: src/sass/_bureau.scss [hunk 2]
```diff
@@ -387,7 +416,27 @@
     }
     tbody {
       border: 1px solid black;
+
+      > div {
+        min-height: 100px;
+      }
```
Comment (elizabeth-jimenez) [inline src/sass/_bureau.scss:422]: image

Should we decrease this min-height so we don't have this extra spacing when we have a single bidder?

**Comment by elizabeth-jimenez** [comment]
Just a few things. The actual ranking functionality works great and looks good! This looks like it was a lot of work 😱 💻

File: src/utilities.test.js [hunk 3]
```diff
@@ -897,3 +898,11 @@ describe('scrollToGlossaryTerm', () => {
...
   });
 });
+
+describe('move', () => {
+  it('moves an item in an array', () => {
+    const arr = [1, 2, 3];
+    const exp = move(arr, 0, 1);
+    expect(exp).toEqual([2, 1, 3]);
+  });
+});
```
Comment (elizabeth-jimenez) [inline src/utilities.test.js:908]: 😍

File: src/actions/bureauPositionBids.js (hunk 1 above)
Comment (mjoyce91) [reply src/actions/bureauPositionBids.js:104]: We could, but the new ranking already appears on the screen

File: src/sass/_bureau.scss (hunk 2 above)
Comment (mjoyce91) [reply src/sass/_bureau.scss:422]: We could - these boxes will always be slightly different in size depending on text length

**Comment by mjoyce91** [comment]
>Scenario: Make an update to the Short List and then Sort. Bidders that you just moved will go back to unRanked. However, if you 'Go Back' and come back into the Position Details, it works as we want.

Okay I see what you mean, I think I know what I need to do to fix this

File: src/sass/_bureau.scss (hunk 2 above)
Comment (elizabeth-jimenez) [reply src/sass/_bureau.scss:422]: yea, some are chonkier lol, but the area 'expands' for those, so ssshould still work the same; no biggie, up to u

Comment (mjoyce91) [reply src/sass/_bureau.scss:422]: We also don't want it too short, otherwise the "hitbox" for dragging a card becomes small

**Comment by mjoyce91** [comment]
@elizabeth-jimenez try again please

File: src/sass/_bureau.scss (hunk 2 above)
Comment (elizabeth-jimenez) [reply src/sass/_bureau.scss:422]: oh for sure, I was thinking just a few pixels, about 10

**Comment by elizabeth-jimenez** [comment]
@mjoyce91 

The sort works now! But the Export is still causing the same issue, tho. 

Also, when we sort by handshake, the shortList is unaffected but we don't get any bidders on the Export. I feel like maybe the shortList should always be exported?

**Comment by mjoyce91** [comment]
@elizabeth-jimenez - see fix for export.

Will probably want to make a new ticket out of the short list comment

**Comment by elizabeth-jimenez** [comment]
> @elizabeth-jimenez - see fix for export.
> 
> Will probably want to make a new ticket out of the short list comment

Yea, good idea

File: src/Components/BureauPage/ExportButton/ExportButton.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react';
```
Comment (elizabeth-jimenez) [inline src/Components/BureauPage/ExportButton/ExportButton.jsx:1]: 🚫  React throwing an Eslint error

**Review by elizabeth-jimenez — APPROVED** [review]
🍰 🍥 🍬 🥮 👍

**Pull request merged**